### DISCUSSION TITLE
Text Reveal Animation

### DISCUSSION THREAD

**Discussion by ankitsharma-99** [fabricjs/fabric.js#9033] [Q&A]
Any ideas on how we can implement text reveal animation on an IText object?

**Comment by ShaMan123** [comment]
Can you be more explicit?
Show a video or something?

**Reply by ShaMan123** [reply]
You can tamper with the rendering methods and have a timer/state to slice the amount but you might want to call `initDimensions`

**Reply by ankitsharma-99** [reply]
I meant something like text reveal effect in premiere pro. https://www.youtube.com/watch?v=r8rhkixuw-0

I am trying to add Rect object as a clipPath to IText object, and then animate the width of the rectangle. However , not much luck on this. 

> You can tamper with the rendering methods and have a timer/state to slice the amount but you might want to call initDimensions

Can you please share an example/code snippet , so that I can use that as a starting point? I am new to fabricjs.

**Reply by ShaMan123** [reply]
A clip path is definitely what you want.
Add a repro, probably you need to mark the itext as dirty so it re-renders the cache
